### PR TITLE
[CI] Add Github actions job to build LLVM documentation

### PR DESCRIPTION
**Pull request by boomanaiden154**
This patch adds in support for building the LLVM documentation through a Github actions job. This enables catching documentation build failures earlier and also more easily as the job failure will show up directly on pull requests. The job currently only builds the documentation for LLVM, but the plan is to extend it to also build the documentation for other subprojects when appropriate (i.e., the docs files have changed), starting with clang.

### CODE REVIEW

**Comment by boomanaiden154** [comment]
I've validated that this works locally with `act` and am interested in reviewer feedback now. Once this lands I'll work on getting a patch up for building the clang docs.

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+jobs:
+  check-docs-build:
+    name: "Test documentation build"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fetch LLVM sources
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
```
Comment (tstellar) [inline .github/workflows/docs.yml:26]: Why 2 here?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# LLVM Documentation CI
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: "Test documentation build"
+
```
Comment (tstellar) [inline .github/workflows/docs.yml:7]:
```suggestion

permissions:
  contents: read
  
```

This is standard for all our workflow files.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (boomanaiden154) [reply .github/workflows/docs.yml]: No particular reason, it was just what @tru used for the clang-format action. I've changed it to one as it shouldn't matter at all for this patch as we only care that the docs build properly at the head commit of the PR/main.

File: .github/workflows/docs.yml (hunk 2 above)
Comment (boomanaiden154) [reply .github/workflows/docs.yml:7]: Good point. Didn't realize that Github by default doesn't give a minimal permissions set.

**Review by tstellar — APPROVED** [review]
LGTM.  Thanks for doing this.

**Comment by boomanaiden154** [comment]
Thanks for the review! I'm going to merge this in a bit and test functionality with a llvm docs patch and start working on support for building the clang documentation tonight.

**Pull request merged**